src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -0,0 +1,453 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+
+namespace Roslyn.Utilities
+{
+    /// <summary>
+    /// The 4.5 portable API surface area does not contain many of the APIs Roslyn needs to function.  In 
+    /// particular it lacks APIs to access the file system.  The Roslyn project though is constrained 
+    /// from moving to the 4.6 framework until post VS 2015.
+    /// 
+    /// This puts us in a difficult position.  These APIs are necessary for us to have our public API set
+    /// in the DLLS we prefer (non Desktop variants) but we can't use them directly when targetting 
+    /// the 4.5 framework.  Putting the APIs into the Desktop variants would create instant legacy for 
+    /// the Roslyn project that we'd have to maintain forever (even if it was just as assemblies with
+    /// only type forward entries).  This is not a place we'd like to be in.  
+    /// 
+    /// As a compromise we've decided to grab these APIs via reflection for the time being.  This is a 
+    /// *very* unfortunate path to be on but it's a short term solution that sets us up for long term
+    /// success.  
+    /// 
+    /// This is an unfortunate situation but it will all be removed fairly quickly after RTM and converted
+    /// to the proper 4.6 portable contracts.  
+    /// </summary>
+    internal static class PortableShim
+    {
+        private const string System_Diagnotsics_FileVersionInfo_Name = "System.Diagnostics.FileVersionInfo, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+        private const string System_IO_FileSystem_Name = "System.IO.FileSystem, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+        private const string System_IO_FileSystem_Primitives_Name = "System.IO.FileSystem.Primitives, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+        private const string System_Runtime_Extensions_Name = "System.Runtime.Extensions, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+        private const string System_Threading_Thread_Name = "System.Threading.Thread, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+
+        /// <summary>
+        /// Find a <see cref="Type"/> instance by first probing the contract name and then the name as it
+        /// would exist in mscorlib.  This helps satisfy both the CoreCLR and Desktop scenarios. 
+        /// </summary>
+        private static Type GetTypeFromEither(string contractName, string desktopName)
+        {
+            Type type;
+            try
+            {
+                type = Type.GetType(contractName, throwOnError: false);
+            }
+            catch (Exception)
+            {
```

Comment:
@davkean that was added when I was trying to track down an issue that eventually proved to be unrelated to my change.  I will remove it as it's not necessary (as you point out).